subprojects/core/src/main/java/org/gradle/api/artifacts/ResolvableDependencies.java

```diff
@@ -60,12 +61,12 @@
     FileCollection getFiles();
 
     /**
-     * Returns a view of this set containing files with the given format.
+     * Returns a view of this set containing files matching the requested attributes.
      *
      * @since 3.3
      */
     @Incubating
-    FileCollection getFiles(String format);
+    FileCollection getFiles(Map<?, ?> attributes);
```

Comment:
Yep, we absolutely require this functionality. It's a way of getting different artifacts for a configuration  _using exactly the same dependency graph_. This is what we're referring to as a "configuration view".

Note that I'm not very happy with this API, but anything else was very hard to use from the consumer build script. I'd be tempted to extract an immutable `Attributes` out of `AttributeContainer`, and then provide a convenient way to construct an `Attributes` instance.